### PR TITLE
Add build with docker info to README

### PR DESCRIPTION
**Pull request by alvicsam**
Added information about how to build and test the code with a docker image that is used in our pipelines. (Related to paritytech/ci_cd#113)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -120,6 +120,27 @@ No native dependencies.
 
 Build with `cargo` and use `ldd` to check that the resulting binary does not depend on OpenSSL anymore.
 
+#### Linux and Docker
+
+Also you can build the repo with 
+[Parity CI Docker image](https://github.com/paritytech/scripts/tree/master/dockerfiles/cachepot-ci):
+
+```bash
+docker pull paritytech/cachepot-ci:staging
```
Comment (drahnr) [inline README.md:129]: Could we replace this by `podman`? It works on at least linux and mac os oob.

File: README.md [hunk 2]
```diff
@@ -120,6 +120,27 @@ No native dependencies.
...
+mkdir ~/cache
+sudo chown 1000:1000 ~/cache #processes in the container runs as "nonroot" user with UID 1000
+docker run --rm -it -w /shellhere/cachepot \
+                    -v /home/$(whoami)/cache/:/cache/    \
+                    -v "$(pwd)":/shellhere/cachepot \
+                    -e CARGO_HOME=/cache/cargo/ \
+                    -e SCCACHE_DIR=/cache/sccache/ \
+                    -e CARGO_TARGET_DIR=/cache/target/  paritytech/cachepot-ci:staging cargo build --locked --release
```
Comment (drahnr) [inline README.md:137]: A few more newlines please :)

File: README.md [hunk 3]
```diff
@@ -120,6 +120,27 @@ No native dependencies.
...
 
+#### Linux and Docker
+
+Also you can build the repo with 
+[Parity CI Docker image](https://github.com/paritytech/scripts/tree/master/dockerfiles/cachepot-ci):
+
+```bash
+docker pull paritytech/cachepot-ci:staging
+mkdir ~/cache
+sudo chown 1000:1000 ~/cache #processes in the container runs as "nonroot" user with UID 1000
```
Comment (drahnr) [inline README.md:131]: 1000 is a bold assumption, `${USER}:${USER}` should be an improvement.

**Review by drahnr — CHANGES_REQUESTED** [review]
A few changes would be required, I am also not convinced this is the right place to place these _parity_ specific instructions, I'd expect those to reside in the project to be built if specifcs are required.

File: README.md (hunk 1 above)
Comment (alvicsam) [reply README.md:129]: Yes, I can, but on Mac the `podman run ... -v ...` command won't work because it still has issues with volume mounting

Comment (alvicsam) [reply README.md:129]: On the other hand, the topic is "Linux and Docker", so I think I'll change it to Podman. In this case it resolves the issue with different UID of host user and container user.

Comment (alvicsam) [reply README.md]: Replaced with `podman`

File: README.md (hunk 2 above)
Comment (alvicsam) [reply README.md]: Fixed

File: README.md (hunk 3 above)
Comment (alvicsam) [reply README.md]: 1000 is not an assumption - this is user id in the `docker.io/paritytech/cachepot-ci:staging` container. So you have to manipulate with this ID and permissions if it differs from user ID.

Comment (drahnr) [reply README.md]: Meta culpa.

In that case we should add the appropriate `--userns=..` option and the :z modifier for selinux to the mapping

**Review by drahnr — APPROVED** [review]
LGTM! Thank you!

**Review by kirushik — APPROVED** [review]

**Pull request merged**